src/containers/SearchPage/SeatsFilter/SeatsFilter.js

```diff
@@ -0,0 +1,162 @@
+      queryParamNames,
+      intl,
+      showAsPopup,
+      ...rest
+    } = this.props;
+
+    const classes = classNames(rootClassName || css.root, className);
+
+    const urlParam = getSeatsQueryParam(queryParamNames);
+    const hasInitialValues = !!initialValues && !!initialValues[urlParam];
+
+    const initialSeats = initialValues[urlParam];
+    const seatsInitialValue = initialSeats ? { seats: initialSeats } : {};
+
+    const labelForPopup = hasInitialValues
+      ? intl.formatMessage({ id: 'SeatsFilter.labelSelected' }, { seatCount: initialSeats })
+      : label;
+
+    const labelForPlain = hasInitialValues ? (
+      <span className={css.labelPlainSelected}>
+        {label}: {initialSeats}
+      </span>
+    ) : (
+      <span className={css.labelPlain}>{label}</span>
+    );
+
+    const handleSubmit = values => {
+      const usedValue = values ? values[name] : values;
+      onSubmit({ [urlParam]: usedValue });
+    };
+
+    // Uncontrolled input needs to be cleared through the reference to DOM element.
+    const handleClear = () => {
+      if (this.mobileInputRef && this.mobileInputRef.current) {
+        this.mobileInputRef.current.value = '';
+      }
+    };
+
+    return showAsPopup ? (
+      <FilterPopup
+        className={classes}
+        rootClassName={rootClassName}
+        popupClassName={css.popupSize}
+        name={name}
+        label={labelForPopup}
+        isSelected={hasInitialValues}
+        id={`${id}.popup`}
+        showAsPopup
+        labelMaxWidth={250}
+        contentPlacementOffset={contentPlacementOffset}
+        onSubmit={handleSubmit}
+        initialValues={seatsInitialValue}
+        keepDirtyOnReinitialize
+        {...rest}
+      >
+        <fieldset className={css.fieldPopup}>
+          <label className={css.fieldPopupLabel} htmlFor={`${id}-input`}>
+            {label}
+          </label>
+          <FieldNumber
+            className={css.field}
+            name={name}
+            id={name}
+            type="text"
+            autoComplete="off"
+            initialValue={initialSeats}
+            minValue={1}
+          />
+        </fieldset>
+      </FilterPopup>
+    ) : (
+      <FilterPlain
+        className={className}
+        rootClassName={rootClassName}
+        label={labelForPlain}
+        isSelected={hasInitialValues}
+        id={`${id}.plain`}
+        liveEdit
+        onSubmit={handleSubmit}
+        onClear={handleClear}
+        initialValues={seatsInitialValue}
+        {...rest}
+      >
+        <fieldset className={css.fieldPlain}>
+          <FieldNumber
+            className={css.field}
+            name={name}
+            id={name}
+            type="text"
+            initialValue={initialSeats}
+            minValue={1}
+          />
+        </fieldset>
+      </FilterPlain>
+    );
+  }
+}
+
+SeatsFilter.defaultProps = {
+  rootClassName: null,
+  className: null,
+  initialValues: null,
+  contentPlacementOffset: 0,
+};
```

Comment:
Let's have a small huddle related to defaultProps.

Reply:
Default props removed and propTypes converted to JSDoc